### PR TITLE
Promoted Content module fixes and improvements

### PR DESCRIPTION
**Pull request by linc**
**Fix `SelectByScore`:** As currently coded, the top-scoring posts will get stuck in the PromotedContent module forever by all-time top scoring. We really just want the freshest content that passed our threshold. Also, the `where` clause was getting double-added.

**Add `SelectByPromoted`:** Add integration with Reactions since it is coming to core anyway.

**Fix conventions & errors**: There were several naming & syntax contention errors preventing the above from working correctly.

### CODE REVIEW

File: applications/vanilla/modules/class.promotedcontentmodule.php [hunk 1]
```diff
@@ -373,6 +377,40 @@ protected function SelectByScore($Parameters) {
...
+      if (!class_exists('ReactionModel')) {
+         return;
+      }
+
+      $RecordTypes = array();
+      if ($this->ShowDiscussions()) {
+         $RecordTypes[] = 'Discussion';
+      }
+      if ($this->ShowComments()) {
+         $RecordTypes[] = 'Comments';
```
Comment (beckyvb) [inline applications/vanilla/modules/class.promotedcontentmodule.php]: This should be $RecordTypes[] = 'Comment';

File: applications/vanilla/modules/class.promotedcontentmodule.php [hunk 2]
```diff
@@ -373,6 +377,40 @@ protected function SelectByScore($Parameters) {
...
+      }
+
+      $ReactionModel = new ReactionModel();
+      $PromotedTagID = $ReactionModel->DefineTag('Promoted', 'BestOf');
+      $Content = $ReactionModel->GetRecordsWhere(
+          array('TagID' => $PromotedTagID, 'RecordType' => $RecordTypes),
+          'DateInserted',
+          'desc',
+          $this->Limit);
+
```
Comment (beckyvb) [inline applications/vanilla/modules/class.promotedcontentmodule.php]: Comments are missing the discussion title.

**Comment by linc** [comment]
Fixt the typo & added a bit that preserves Name for comments when it's already set (as Reactions does).

**Comment by beckyvb** [comment]
Looks good!

**Pull request merged**